canonical/maas-ui
# feat(vault): Add incomplete setup warning to machine list

## Done

- Add warning message to machine list if not all region controllers are configured with Vault
- Add warning message to machine list if region controllers are configured with Vault but secrets are not migrated

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /machines
- Ensure no Vault warnings are displayed

## Fixes

Fixes https://github.com/canonical/app-tribe/issues/1407

## Screenshots

### Not all controllers configured
![image](https://user-images.githubusercontent.com/35104482/196670383-688faaad-33af-4af5-973a-56c325073139.png)

### Secrets not migrated
![image](https://user-images.githubusercontent.com/35104482/196668308-4288aa3e-17e9-4388-a758-34cfb4044f25.png)
